benchexec/container.py

```diff
@@ -812,6 +861,32 @@ def drop_capabilities(keep=[]):
     )
 
 
+def cap_permitted_to_ambient():
+    """
+    Python version of util-linux/lib/caputils.c: cap_permitted_to_ambient()
+    """
+
+    header = libc.CapHeader(libc.LINUX_CAPABILITY_VERSION_3, 0)
+    data = (libc.CapData * libc.LINUX_CAPABILITY_U32S_3)()
+
+    libc.capget(header, data)
+
+    data[0].inheritable = data[0].permitted
+    data[1].inheritable = data[1].permitted
+
+    libc.capset(header, data)
+
+    effective = (data[1].effective << 32) | data[0].effective
+    for cap in range(64):
+        if cap > int(util.try_read_file("/proc/sys/kernel/cap_last_cap")):
```

Comment:
This makes us open and read the file in every loop iteration. We could just read it once and use it as upper bound of `range`. (Make sure to add 1 to make the bound inclusive.)